paper-drawer-panel.html

```diff
@@ -750,16 +770,54 @@
           if (translateX === null) {
             return '';
           }
-
           return this.hasWillChange ? 'translateX(' + translateX + 'px)' :
               'translate3d(' + translateX + 'px, 0, 0)';
         },
 
         _moveDrawer: function(translateX) {
           this.transform(this._transformForTranslateX(translateX), this.$.drawer);
-        }
+        },
 
+        _getDrawerContent: function() {
+          return Polymer.dom(this.$.drawerContent).getDistributedNodes()[0];
+        },
+
+        _getAutoFocusedNode: function() {
+          var drawerContent = this._getDrawerContent(); 
+          return Polymer.dom(drawerContent).querySelector(this.drawerFocusSelector) || drawerContent;
+        },
+
+        _toggleFocusListener: function(selected) {
+          if (selected === 'drawer') {
+            this._focusedChild = this._getAutoFocusedNode();
+            document.addEventListener('focus', this._boundFocusListener, true);
+          } else {
+            this._focusedChild = null;
+            document.removeEventListener('focus', this._boundFocusListener, true);
+          }
+        },
+
+        _didFocus: function(event) {
+          var path = Polymer.dom(event).path;
+          var focusedChild = path[0];
+          var drawerContent = this._getDrawerContent();
+          var focusedChildCameFromDrawer = false;
+
+          while (!focusedChildCameFromDrawer && path[0] && path[0].hasAttribute) {
+            focusedChildCameFromDrawer = path.shift() === drawerContent;
+          }
+          if (focusedChildCameFromDrawer) {
+            this._focusedChild = focusedChild;
+          } else if (this._focusedChild) {
+            this._focusedChild.focus();
```

Comment:
In the demo, if I'm focused to the last element in the drawer (`this._focusedChild == [Close drawer button]`), then tab to the address bar, then tab again, my focus is restored to the [Close drawer button], even though I expect to be back at the top (i.e. [Options button]).

I think the solution to this is two-fold. First, we should modify the default `drawerFocusSelector` to include anchors, buttons, and inputs (e.g. `[tabindex], a, button, input`) so that we can find the [Options button] in the demo. Second, we should just focus to `this._getAutoFocusedNode()` instead of keeping track of `this._focusedChild`

Also, as soon as the drawer is opened, we should immediately focus to `this._getAutoFocusedNode()` (i.e. the toggle button should lose focus, since it should now be covered by the drawer/scrim.